### ISSUE TITLE
Get-Help / -? should show the parameter-set name above each syntax diagram

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#6856]
<sup>Related: #4135, #4715, #4626, #4137</sup>

Especially with complex commands that have several parameter sets, the parameter-set names are important to quickly understand the context of a given syntax diagram.

The help-topic sources seems to already contain the information, but it shows neither with `Get-Help` / `-?` nor online - see https://github.com/PowerShell/PowerShell-Docs/issues/2422

A simple example:

Steps to reproduce
------------------

```powershell
Get-Content -?
```

Expected behavior
-----------------

```none
...
    
SYNTAX
  Path (Default)

    Get-Content [-Encoding {Unknown | String | Unicode | Byte | BigEndianUnicode | UTF8 | UTF7 | UTF32 | Ascii | Default | Oem | BigEndianUTF32}] [-Delimiter <String>] [-Force] [-Raw] [-Wait] [-Stream <String>] [-UseTransaction] [-Credential 
    <PSCredential>] [-Exclude <String[]>] [-Filter <String>] [-Include <String[]>] -LiteralPath <String[]> [-ReadCount <Int64>] [-TotalCount <Int64>] [-Tail <Int32>] [<CommonParameters>]

  LiteralPath
    
    Get-Content [-Path] <String[]> [-Encoding {Unknown | String | Unicode | Byte | BigEndianUnicode | UTF8 | UTF7 | UTF32 | Ascii | Default | Oem | BigEndianUTF32}] [-Delimiter <String>] [-Force] [-Raw] [-Wait] [-Stream <String>] [-UseTransaction] 
    [-Credential <PSCredential>] [-Exclude <String[]>] [-Filter <String>] [-Include <String[]>] [-ReadCount <Int64>] [-TotalCount <Int64>] [-Tail <Int32>] [<CommonParameters>]
    
...
```

Actual behavior
---------------

```none
...
    
SYNTAX
    Get-Content [-Encoding {Unknown | String | Unicode | Byte | BigEndianUnicode | UTF8 | UTF7 | UTF32 | Ascii | Default | Oem | BigEndianUTF32}] [-Delimiter <String>] [-Force] [-Raw] [-Wait] [-Stream <String>] [-UseTransaction] [-Credential 
    <PSCredential>] [-Exclude <String[]>] [-Filter <String>] [-Include <String[]>] -LiteralPath <String[]> [-ReadCount <Int64>] [-TotalCount <Int64>] [-Tail <Int32>] [<CommonParameters>]
    
    Get-Content [-Path] <String[]> [-Encoding {Unknown | String | Unicode | Byte | BigEndianUnicode | UTF8 | UTF7 | UTF32 | Ascii | Default | Oem | BigEndianUTF32}] [-Delimiter <String>] [-Force] [-Raw] [-Wait] [-Stream <String>] [-UseTransaction] 
    [-Credential <PSCredential>] [-Exclude <String[]>] [-Filter <String>] [-Include <String[]>] [-ReadCount <Int64>] [-TotalCount <Int64>] [-Tail <Int32>] [<CommonParameters>]
    
...
```

Environment data
----------------

Written as of:

```powershell
PowerShell Core v6.0.2
```

**Comment by BrucePay** [comment]
FYI: parameter set names are not considered to be user-facing data and there is no expectation that they should be sensible to the end-user. They're only intended to be used by the command developer and changing a parameter set name is not considered to be a breaking change. 

In practice, they're mostly some amalgam of the distinguishing parameter names, sometimes with "Set" on the end so it might be useful information in some (maybe most) cases.  But, as proposed,  it also takes up a lot of space. It might be more effective to show the syntax with the distinguishing parameters first instead of displaying them all alphabetically.

**Comment by mklement0** [comment]
@BrucePay: 

> It might be more effective to show the syntax with the distinguishing parameters first instead of displaying them all alphabetically.

That would certainly help. Ultimately it is about being able to quickly discern what's unique about a parameter set.

> changing a parameter set name is not considered to be a breaking change.

Understood, but for the purpose of quickly _informing_ users (as opposed to _programmatic processing_) that in itself is not a problem.

If chosen carefully - and I understand that that is not a current requirement - a parameter-set name can potentially communicate more than the just the name of the unique parameter.

> it also takes up a lot of space.

True, but a condensed display doesn't help if it hinders readability.

This is a bit of a tangent, but: 

I find the current syntax diagrams hard to parse visually in general.

Consider the more "spacious" but far more legible representation @rkeithhill mentions in https://github.com/PowerShell/PowerShell/issues/4135#issuecomment-311788186.

**Comment by fowl2** [comment]
> FYI: parameter set names are not considered to be user-facing data

FWIW, they _are_ actually displayed by the ISE - each tab is a parameter set:

![image](https://user-images.githubusercontent.com/12844147/61267188-3414d380-a7da-11e9-860f-2f6c4387de2c.png)
